storage/innobase/include/ut0new.h

```diff
@@ -357,13 +404,17 @@ class ut_allocator {
 			}
 		}
 
+
+
 #ifdef UNIV_PFS_MEMORY
 		ut_new_pfx_t*	pfx = static_cast<ut_new_pfx_t*>(ptr);
 
-		allocate_trace(total_bytes, file, pfx);
+		ut_dontdump(ptr, total_bytes, dontdump, pfx, file);
 
 		return(reinterpret_cast<pointer>(pfx + 1));
 #else
+		ut_dontdump(ptr, total_bytes, dontdump, NULL, file);
```

Comment:
If we call `ut_dontdump()` with `pfx=NULL`, how can the `ut_dodump()` be called when the memory is freed? `ut_dodump()` always relies on `pfx` to determine the size.